broker-core/src/test/java/io/zeebe/broker/job/JobStateControllerTest.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Zeebe Broker Core
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.zeebe.broker.job;
+
+import static io.zeebe.util.buffer.BufferUtil.wrapString;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.zeebe.broker.job.JobStateController.State;
+import io.zeebe.protocol.impl.record.value.job.JobRecord;
+import io.zeebe.util.buffer.BufferUtil;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.agrona.DirectBuffer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class JobStateControllerTest {
+  @Rule public TemporaryFolder folder = new TemporaryFolder();
+
+  private JobStateController stateController;
+
+  @Before
+  public void setUp() throws Exception {
+    stateController = new JobStateController();
+    stateController.open(folder.newFolder("db"), false);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    stateController.close();
+  }
+
+  @Test
+  public void shouldCreateJobEntry() {
+    // given
+    final long key = 1L;
+    final JobRecord jobRecord = createJobRecord();
+
+    // when
+    stateController.create(key, jobRecord);
+
+    // then
+    assertThat(stateController.exists(key)).isTrue();
+    assertJobState(key, State.ACTIVATABLE);
+    assertJobRecordIsEqualTo(stateController.getJob(key), jobRecord);
+  }
+
+  @Test
+  public void shouldActivateJob() {
+    // given
+    final long key = 1L;
+    final JobRecord jobRecord = createJobRecord();
+
+    // when
+    stateController.activate(key, jobRecord);
```

Comment:
Can I activate a job which is already activated?

Reply:
Yes, and if we don't add any pre-conditions check in the state, then the result is that it will overwrite the previous operations, so essentially update the deadline I guess.